puma/hlplots/tagger.py

```diff
@@ -224,3 +259,71 @@ def discriminant(self, signal: Flavour, fx: float | None = None):
             sig_var = self.variables[self.output_flavours.index(Flavours[signal])]
             return self.scores[sig_var]
         raise ValueError(f"No discriminant defined for {signal} signal.")
+
+    def vertex_indices(self, incl_vertexing=False):
+        """Retrieve cleaned vertex indices for the tagger.
+
+        Parameters
+        ----------
+        incl_vertexing : bool, optional
+            Whether to merge all vertex indices, by default False.
+
+        Returns
+        -------
+        np.ndarray
+            Vertex indices for the tagger
+        """
+        if "vertexing" not in self.aux_tasks:
+            raise ValueError("Vertexing aux task not available for this tagger.")
+        else:
+            truth_indices = self.aux_labels["vertexing"]
+            reco_indices = self.aux_scores["vertexing"]
+
+        # clean truth vertex indices - remove indices from true PV, PU, fake
+        truth_removal_cond = np.logical_or(
+            self.aux_labels["vertexing"] == 0,
+            np.isin(self.aux_labels["track_origin"], [0, 1, 2]),
+        )
+        truth_indices = clean_indices(
+            truth_indices,
+            truth_removal_cond,
+            mode="remove",
+        )
+
+        # merge truth vertices from HF for inclusive performance
+        if incl_vertexing:
```

Comment:
I might have missed it but can we also add some text to the `atlas_second_tag` to mark plots as exclusive/inclusive?

Reply:
The atlas_second_tag is completely blank by default I think so I just decided to leave it that way for these plots and let the user specify what they want. I can write a default second_tag if that would be useful though.

Reply:
Yeah when you instantiate the plots would be good to automatically add a string for inclusive/exclusive. Otherwise it won't be easy to differentiate the two

Reply:
Done

Reply:
thanks!